# HG changeset patch
# User ttl <ttl@justmail.de>
# Date 1312020642 -7200
#      Sat Jul 30 12:10:42 2011 +0200
# Node ID 0d18b0e168ae3398f1a82208dbba6598817933d0
# Parent  be425f45f9e35143396b82adc2f92ce72a07ef2a
editor: shortcuts for toolbar actions, window icon

diff --git a/gui/src/FileEditorMdiSubWindow.cpp b/gui/src/FileEditorMdiSubWindow.cpp
--- a/gui/src/FileEditorMdiSubWindow.cpp
+++ b/gui/src/FileEditorMdiSubWindow.cpp
@@ -188,17 +188,17 @@ FileEditorMdiSubWindow::saveFile (QStrin
 
 void
 FileEditorMdiSubWindow::saveFileAs ()
 {
   QString saveDir(m_fileName);
   if (saveDir==UNNAMED_FILE)
     saveDir = QDir::homePath();
   QString saveFileName = QFileDialog::getSaveFileName(
-        this, "Save File", saveDir,SAVE_FILE_FILTER);
+        this, "Save File As", saveDir,SAVE_FILE_FILTER);
   if(saveFileName.isEmpty())
     return;
   saveFile(saveFileName);
 }
 
 // TODO: Do we still need tool tips in the status bar? Tool tips are now
 //       shown directly at the theme icons
 void
@@ -283,54 +283,64 @@ FileEditorMdiSubWindow::construct ()
         QIcon::fromTheme("document-new",style->standardIcon (QStyle::SP_FileIcon)),
         tr("&New File"), m_toolBar);
   QAction *openAction = new QAction (
         QIcon::fromTheme("document-open",style->standardIcon (QStyle::SP_DirOpenIcon)),
         tr("&Open File"), m_toolBar);
   QAction *saveAction = new QAction (
         QIcon::fromTheme("document-save",style->standardIcon (QStyle::SP_DriveHDIcon)),
         tr("&Save File"), m_toolBar);
-  QAction *saveActionAs = new QAction (
+  QAction *saveAsAction = new QAction (
         QIcon::fromTheme("document-save-as",style->standardIcon (QStyle::SP_DriveFDIcon)),
         tr("Save File &As"), m_toolBar);
   QAction *undoAction = new QAction (
         QIcon::fromTheme("edit-undo",style->standardIcon (QStyle::SP_ArrowLeft)),
         tr("&Undo"), m_toolBar);
   QAction *redoAction = new QAction (
         QIcon::fromTheme("edit-redo",style->standardIcon (QStyle::SP_ArrowRight)),
         tr("&Redo"), m_toolBar);
 
+  // short cuts
+  closeAction->setShortcut(QKeySequence::Close);
+  newAction->setShortcut(QKeySequence::New);
+  openAction->setShortcut(QKeySequence::Open);
+  saveAction->setShortcut(QKeySequence::Save);
+  saveAsAction->setShortcut(QKeySequence::SaveAs);
+  undoAction->setShortcut(QKeySequence::Undo);
+  redoAction->setShortcut(QKeySequence::Redo);
+
   m_toolBar->addAction (closeAction);
   m_toolBar->addAction (newAction);
   m_toolBar->addAction (openAction);
   m_toolBar->addAction (saveAction);
-  m_toolBar->addAction (saveActionAs);
+  m_toolBar->addAction (saveAsAction);
   m_toolBar->addAction (undoAction);
   m_toolBar->addAction (redoAction);
 
   QVBoxLayout *layout = new QVBoxLayout ();
   layout->addWidget (m_toolBar);
   layout->addWidget (m_editor);
   layout->addWidget (m_statusBar);
   layout->setMargin (2);
   widget ()->setLayout (layout);
 
   connect (closeAction, SIGNAL (triggered()), this, SLOT (close()));
   connect (newAction, SIGNAL (triggered ()), this, SLOT (newFile ()));
   connect (openAction, SIGNAL (triggered ()), this, SLOT (openFile ()));
   connect (undoAction, SIGNAL (triggered ()), m_editor, SLOT (undo ()));
   connect (redoAction, SIGNAL (triggered ()), m_editor, SLOT (redo ()));
   connect (saveAction, SIGNAL (triggered ()), this, SLOT (saveFile ()));
-  connect (saveActionAs, SIGNAL (triggered ()), this, SLOT (saveFileAs ()));
+  connect (saveAsAction, SIGNAL (triggered ()), this, SLOT (saveFileAs ()));
 
   // TODO: Do we still need tool tips in the status bar? Tool tips are now
   //       shown directly at the theme icons
   connect (newAction, SIGNAL (hovered ()), this, SLOT (showToolTipNew ()));
   connect (openAction, SIGNAL (hovered ()), this, SLOT (showToolTipOpen ()));
   connect (undoAction, SIGNAL (hovered ()), this, SLOT (showToolTipUndo ()));
   connect (redoAction, SIGNAL (hovered ()), this, SLOT (showToolTipRedo ()));
   connect (saveAction, SIGNAL (hovered ()), this, SLOT (showToolTipSave ()));
-  connect (saveActionAs, SIGNAL (hovered ()), this,SLOT (showToolTipSaveAs ()));
+  connect (saveAsAction, SIGNAL (hovered ()), this,SLOT (showToolTipSaveAs ()));
 
   m_fileName = "";
   setWindowTitle (m_fileName);
+  setWindowIcon(QIcon::fromTheme("accessories-text-editor",style->standardIcon (QStyle::SP_FileIcon)));
   show ();
 }
